### DISCUSSION TITLE
ValueError: invalid literal for int() with base 10: ''

### DISCUSSION THREAD

**Discussion by qn222ad** [highfestiva/finplot#198] [Q&A]
Can I ask what is the error of this one and how should I solve them?

ValueError: invalid literal for int() with base 10: ''
```
These are my data type:
Open      float64
High      float64
Low       float64
Close     float64
Volume      int64
Time        int64 #this one is doing for onclick and take date
the index is datetime index I checked
`if data.index.__class__.__name__ == 'DatetimeIndex': print('true')`

My read function
```python
def read_days_only(df):
    dfn = df.copy()
    dfn.dropna(inplace=True)
    dfn.loc[:,'Date'] = pd.to_datetime(dfn['Date'], errors='coerce',format='%Y-%m-%d',utc=True)
    dfn = dfn.set_index(pd.DatetimeIndex(dfn['Date']))
    return dfn
data = read_days_only(pd.read_csv('data/dfn_dji.csv'))
ax = fplt.create_plot('pivot', rows=1)
fplt.candlestick_ochl(data[['Open','Close','High','Low']],ax=ax)
fplt.show()
```

**Comment by qn222ad** [comment]
I found problem was that my data is from 1928 or older that why in pandas used default since 1970. 
I set the date as
    dfn['Date'] = pd.to_datetime([x for x in range(23823)], unit='D',origin=pd.Timestamp('1928-10-02'))
but it did not work too... now, I do not know how to fix this

**Reply by highfestiva** [reply]
You'll have to ask the Pandas folks about that one.

**Reply by qn222ad** [reply]
I should do a search how tradingview can plot they chart from 1897

**Reply by qn222ad** [reply]
Thank you! I like finplot so much, it is very fast and beautiful. compared to plotly and matplotlib, yours library is a real beauty. Also thank you very much for always give me helpfull feedback

**Reply by highfestiva** [reply]
Not always. Thanks for you kind words, you get a fix, using e93c2ca you can display UTC dates prior to 1970. :)

**Reply by qn222ad** [reply]
Yeah.. I can plot them now. Thank you very much! Can I ask after I plotted lines by add_line function, what should I do if I want to click and remove lines?

**Reply by highfestiva** [reply]
Call  `remove_primitive()`.

**Reply by qn222ad** [reply]
Thank you:)

**Reply by qn222ad** [reply]
Can I ask, if I want to plot candlestick chart but not for datetime format but for the date in a year ( from 1 to 365 or 366), where can safety change so that I won't affect to other code? :) I want to view data in a form of 1 to 365 so that I can compare overlay data for each year:D

**Reply by highfestiva** [reply]
Reset the index, and use 0 for first year, 1 for second, etc. Then plot. Do not use dates at all. It helps if you know what you're doing in pandas.
```python
import yfinance as yf
import finplot as fplt

df = yf.download('TSLA', '2020-12-30')
df = df.reset_index(drop=True)
df = df.reset_index()
cols = ['num'] + list(df.columns[1:])
df.columns = cols
fplt.candlestick_ochl(df[['num','Open','Close','High','Low']])
fplt.show()
```

**Reply by qn222ad** [reply]
Thank you:)

**Reply by qn222ad** [reply]
I had this error again:
ValueError: invalid literal for int() with base 10: ''
while yesterday, I used the same function, it plotted. today I rerun, I got the error :(:(. now I do not know why it had the error. all of my previous plot, work fine now had same error,
please give me some suggestions !!
Thanks

my code:
df2[f'{col}'] = df2['Close'][pd.to_datetime(df2.index.date.astype(str)).isin(pd.to_datetime(df_count_all.index.date))]
 fplt.candlestick_ochl(df2[['Open', 'Close', 'High', 'Low']])
fplt.plot(df2[f'{col}'], color='#1ad845', style='^', legend=f'all aspect in 30 set of {col}', width=3)
fplt.show()
_the most strange things is it works fine previously. I did not even update my pycham....._
the data
[data.zip](https://github.com/highfestiva/finplot/files/6903523/data.zip)
I plot the mercury collumn as symbol in candlestick chart

**Reply by highfestiva** [reply]
Please post executable code and data in one zip.

**Reply by qn222ad** [reply]
Please find it here
[plot.zip](https://github.com/highfestiva/finplot/files/6906991/plot.zip)
Thanks!

**Reply by highfestiva** [reply]
`ModuleNotFoundError: No module named 'aspect_extraction'`

**Reply by qn222ad** [reply]
I forgot to delete that line, can you delete that line, the code is complete without that line
Thanks

**Reply by qn222ad** [reply]
import finplot as fplt
import pandas as pd

data = read_days_only(pd.read_csv('data/dfn-dji.csv'))
pivot = read_days_only(pd.read_csv('data/df_sq.csv'))
pivot=pivot.sort_index()

def read_days_only(df):
    dfn = df.copy()
    dfn.loc[:,'Date'] = pd.to_datetime(dfn['Date'], errors='coerce')
    dfn = dfn.set_index(pd.DatetimeIndex(dfn['Date'].dt.date))
    return dfn
data[f'sq'] = data['Low'][pd.to_datetime(data.index.date.astype(str)).isin(pd.to_datetime(pivot.index.date))]
fplt.candlestick_ochl(data[['Open', 'Close', 'High', 'Low']])
fplt.plot(data[f'sq'], color='#1ad845', style='^', legend=f'square', width=3)

fplt.show()

**Reply by highfestiva** [reply]
Dunno, works for me. Fetch the latest finplot code and see if that works.

**Reply by qn222ad** [reply]
I reinstall several times it is still the same. Previously, code run just fine, now all the previous code file, same error...strange..

**Reply by highfestiva** [reply]
Could be that a necessary patch is not yet released. Try to fetch the latest code here from github and replace your locally installed `site-packages/finplot/__init__.py` with that.

**Reply by qn222ad** [reply]
:):) yes, replaced the file works :):) thanks!!!!
